### PR TITLE
Feature/22 autorization back

### PR DESCRIPTION
**Pull request by Kosichenko**
All done.

### CODE REVIEW

File: slidez-core/src/main/java/com/binarystudio/academy/slidez/domain/user/UserService.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    private User convertUser(UserDto userDto) {
+        return User.builder()
+                .email(userDto.getEmail())
+                .password(passwordEncoder.encode(userDto.getPassword()))
+                .firstName(userDto.getFirstName())
+                .lastName(userDto.getLastName())
+                .build();
+    }
```
Comment (Anatolii-Ivanov) [inline slidez-core/src/main/java/com/binarystudio/academy/slidez/domain/user/UserService.java:42]: Move it to a separate mapper out of UserService

File: slidez-core/src/main/java/com/binarystudio/academy/slidez/domain/user/dto/UserDetailsDto.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+import lombok.Builder;
+import lombok.Data;
+
+import java.util.UUID;
+
+@Data
+@Builder
+public class UserDetailsDto {
+        private String email;
+        private UUID id;
```
Comment (Anatolii-Ivanov) [inline slidez-core/src/main/java/com/binarystudio/academy/slidez/domain/user/dto/UserDetailsDto.java:12]: Move id to first place

File: slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthController.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	private final AuthService authService;
+	private final PasswordValidator passwordValidator;
+	private final EmailValidator emailValidator;
+
+	@Autowired
+	public AuthController(AuthService authService, PasswordValidator passwordValidator, EmailValidator emailValidator) {
+		this.authService = authService;
+		this.passwordValidator = passwordValidator;
+		this.emailValidator = emailValidator;
+	}
```
Comment (Anatolii-Ivanov) [inline slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthController.java:28]: Can't we use it this way?
    @Autowired
    private AuthService authService;

File: slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthController.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	@PostMapping("login")
+	public ResponseEntity login(@RequestBody AuthorizationRequest authorizationRequest) {
+		if(!passwordValidator.isValid(authorizationRequest.getPassword())) {
+			return new ResponseEntity<>("Incorrect password", HttpStatus.BAD_REQUEST);
+		}
+
+		if(!emailValidator.isValid(authorizationRequest.getEmail())) {
+			return new ResponseEntity<>("Incorrect email", HttpStatus.BAD_REQUEST);
+		}
+
```
Comment (Anatolii-Ivanov) [inline slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthController.java:39]: You repeat this code, create a func which will get email + password and then returns error message and validation result

File: slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthService.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	public AuthService(UserService userService, JwtProvider jwtProvider, PasswordEncoder passwordEncoder) {
+		this.userService = userService;
+		this.jwtProvider = jwtProvider;
+		this.passwordEncoder = passwordEncoder;
+	}
+
+	public AuthResponse performLogin(AuthorizationRequest authorizationRequest)  {
+		var userOptional = userService.findByEmail(authorizationRequest.getEmail());
+		User user = userOptional
+				.orElseThrow(() ->new ResponseStatusException(HttpStatus.UNAUTHORIZED, "User not found"));
```
Comment (Anatolii-Ivanov) [inline slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthService.java:35]: It's service, it shouldn't throw HTTPStatus exceptions

Comment (Anatolii-Ivanov) [reply slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthService.java:35]: Everywhere in this service

File: slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthService.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+	}
+
+	public UserDetailsDto mapUserUserDetailsDto(User user) {
+		return   UserDetailsDto.builder()
+				.id(user.getId())
+				.email(user.getEmail())
+				.firstName(user.getFirstName())
+				.lastName(user.getLastName())
+				.build();
+	}
```
Comment (Anatolii-Ivanov) [inline slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthService.java:64]: To the separate file

Comment (Anatolii-Ivanov) [reply slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/AuthService.java:64]: Or userDetails model

File: slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/model/AuthResponse.java [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+public class AuthResponse {
+	private String accessToken;
+	private UserDetailsDto userDetailsDto;
+
+	public static AuthResponse of(String accessToken, UserDetailsDto userDetailsDto) {
+		AuthResponse response = new AuthResponse();
+		response.setAccessToken(accessToken);
+		response.setUserDetailsDto(userDetailsDto);
+		return response;
+	}
```
Comment (Anatolii-Ivanov) [inline slidez-core/src/main/java/com/binarystudio/academy/slidez/infrastructure/security/auth/model/AuthResponse.java:16]: Use something like 

@NoArgsConstructor
@AllArgsConstructor

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by annaprusakova — APPROVED** [review]

**Review by Anatolii-Ivanov — APPROVED** [review]

**Pull request merged**